group/assignment5/UnknownAlphabet.java

```diff
@@ -1,194 +1,266 @@
-
-		cur_vertex.setState("grey"); // starting visit, but not finished
-		for (Vertex child : cur_vertex.getChildren()) {
-			if (child.getState().equals("white")) {
-				sortFromVertex(child);
-			} else if (child.getState().equals("grey")) {
-				throw new Exception("There is a cycle, order impossible.");
-			}
-		}
-		cur_vertex.setState("black"); // finished visit.
-		alphabet.addFirst(cur_vertex.getValue());
-	}
-
-	// Look at every 2 consecutive words to get the relation between the word_first 2 different characters.
-	// For example, in [ABCD, ABDE, EFGH, FMNO]: C < D (from word_first 2 words), A < E (from 2nd and 3rd), E < F (from 3rd and 4th)
-	// Because A < E and E < F then from transitivity => A < E without needing to check word 1 or 2 with word 4.
-	public void createVertices(List<String> dict) {
-
-		boolean isSecondWordLast = false;
-		int dictSize = dict.size();
-		
-		for (int i = 0; i < dictSize - 1; i++) {
-			String word_first = dict.get(i);
-			String word_second = dict.get(i + 1);
-			
-			if(i == dictSize - 1)
-				isSecondWordLast = true; 
-			
-			createVerticesHelper(word_first, word_second, isSecondWordLast);
-		}
-	}
-
-	private void createVerticesHelper(String word_first, String word_second, boolean isSecondWordLast) {
-
-		int length_first = word_first.length();
-		int length_second = word_second.length();
-		int min_length = Math.min(length_first, length_second);
-
-		int index = 0;
-
-		// Iterate through word_first and word_second word until they have a different character.
-		// For each new character, create its vertex and add it to the list using createVertex.
-		while (index < min_length) {
-			char char_first = word_first.charAt(index);
-			char char_second = word_second.charAt(index);
-
-			if (char_first == char_second) {
-				createVertex(char_first);
-			}
-
-			else {
-				Vertex vertex_first = createVertex(char_first);
-				Vertex vertex_second = createVertex(char_second);
-				vertex_first.addChild(vertex_second);
-				break;
-			}
-
-			index++;
-		}
-
-		int index_second = index;
-		
-		// Create vertices for characters until the end of word_first  word.
-		while (index < length_first) {
-			createVertex(word_first.charAt(index));
-			index++;
-		}
-
-		if (isSecondWordLast) {
-			while (index_second < length_second) {
-				createVertex(word_second.charAt(index_second));
-				index_second++;
-			}
-		}
-
-	}
-			
-	// Auxiliary method, keep a map of already created vertices for given chars.
-	// Only create new vertex and update map and list if char not seen before (not in map).
-	private Vertex createVertex(char ch) {
-		
-		Vertex vertex;
-
-		if (!char_vertex_map.containsKey(ch)) {
-			vertex = new Vertex(ch);
-			char_vertex_map.put(ch, vertex);
-			vertices.add(vertex);
-		} else {
-			vertex = char_vertex_map.get(ch);
-		}
-
-		return vertex;
-	}
-	
-	// Getter method for testing.
-	public LinkedList<Vertex> getListOfVertices(){
-		return vertices;
-	}
-
-}
+import java.util.HashMap;

```

Comment:
Ana, thank you for extending the solution with the challenge and improved it with camelCase. I've had a quick look and I don't have anything to add. If no one else has other comments, feel free to pull it to master.